### ISSUE TITLE
Add support for ConstExpr

### ISSUE THREAD

**Issue by antonmedv** [antonmedv/expr#84]
Add support for constexpr:

```go
expr.Compile("foo(...)", expr.Env(env{}), expr.ConstExpr("foo"))
```

If helper declared as constexpr with `expr.ConstExpr(string)`, it (helper) should be treated as a pure function. Compile will evaluate it during compile and replace it with the result.

**Comment by antonmedv** [comment]
@davidklassen

**Comment by antonmedv** [comment]
Implemented in #96
